lib/web_ui/lib/src/engine/text/paragraph.dart

```diff
@@ -36,6 +40,15 @@ class EngineLineMetrics implements ui.LineMetrics {
   /// The textual content representing this line.
   final String text;
 
+  /// The index in the text where this line begins.
+  final int startIndex;
+
+  /// The index in the text where this line ends.
```

Comment:
Age-old question: is that the index of the last character or +1 from that? :)

Reply:
It's `[start, end[`
i.e. `start` is inclusive, `end` is exclusive.

I'll clarify it in the comment.